### PR TITLE
Feature/create custom policy class

### LINKED ISSUE #2: Create a custom Policy class
Create a custom `PolicyGraph` that receives a configuration dict and defines actions to take, a trajectory postprocessor for experiences, and a loss function.

### PR DESCRIPTION
**Pull request by angelolovatto**
Should close #2 and eventually #13

### CODE REVIEW

File: mapo/agents/mapo/mapo_policy.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+    # Use sigmoid to scale to [0,1], but also double magnitude of input to
+    # emulate behaviour of tanh activation used in DDPG and TD3 papers.
+    sigmoid_out = tf.nn.sigmoid(2 * action_out)
+    # Rescale to actual env policy scale
+    # (shape of sigmoid_out is [batch_size, dim_actions], so we reshape to
+    # get same dims)
+    action_range = (action_space.high - action_space.low)[None]
+    low_action = action_space.low[None]
+    actions = action_range * sigmoid_out + low_action
```
Comment (thiagopbueno) [inline mapo/agents/mapo/mapo_policy.py]: It should be noted that using the sigmoid function this way to force the action values to remain between the bounds of the `action_space` has an multiplicative effect on the gradient of the activation function. In principle, this can be troublesome depending on the scale of the `action_range`...

One possible solution is to also divide the argument given the the sigmoid function by `action_range`. So the gradient of the modified sigmoid remais the same of the original sigmoid, even though the range is now given by `action_range`.

File: mapo/agents/mapo/mapo_policy.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        state_preprocessor = None
+        action_out = input_dict[SampleBatch.CUR_OBS]
+
+    activation = config["actor_hidden_activation"]
+    for hidden in config["actor_hiddens"]:
+        action_out = keras.layers.Dense(units=hidden, activation=activation)(action_out)
+    action_out = keras.layers.Dense(units=action_space.shape[0], activation=None)(
+        action_out
+    )
+
```
Comment (thiagopbueno) [inline mapo/agents/mapo/mapo_policy.py]: When we refactor this code, I think it would be interesting to extend `tf.keras.layers.Layer` to define our own `ActionBoundedLayer` so to encapsulate this squashing function (or even to implement other approaches, e.g., clipping, ...). 

That way we could have a proper `tf.keras.models.Sequential` model here the same way function `build_continuous_q_function` builds the critic network...

For this initial version this is fine...

File: mapo/agents/mapo/mapo_policy.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    with tf.compat.v1.variable_scope(Q_SCOPE):
+        policy.q_values, policy.q_preprocessor = build_continuous_q_function(
+            policy.get_obs_input_dict(),
+            policy.observation_space,
+            policy.action_space,
+            policy.config,
+        )
+
+    # Fitted Q loss (using trajectory returns)
+    critic_loss = tf.reduce_mean(
```
[the remark this answers is not in the document]
Comment (angelolovatto) [reply mapo/agents/mapo/mapo_policy.py]: I agree!
